scaladoc/src/dotty/tools/scaladoc/DocContext.scala

```diff
@@ -73,10 +73,24 @@ case class NavigationNode(name: String, dri: DRI, nested: Seq[NavigationNode])
 
 case class DocContext(args: Scaladoc.Args, compilerContext: CompilerContext):
   lazy val sourceLinks = SourceLinks.load(args.sourceLinks, args.revision)(using compilerContext)
+
+  lazy val snippetCompilerArgs = snippets.SnippetCompilerArgs.load(args.snippetCompiler, args.snippetCompilerDebug)(using compilerContext)
+
+  lazy val snippetChecker = snippets.SnippetChecker(
+    args.classpath, 
+    args.bootclasspath, 
+    args.tastyFiles, 
+    compilerContext.settings.scalajs.value(using compilerContext),
+    compilerContext.settings.usejavacp.value(using compilerContext)
+  )
```

Comment:
There might be more settings needed to compile snippets (for example, if a compiler plugin is used), and adding them one by one isn't going to scale, could you pass all of compilerContext.settings instead?

Reply:
Yes, you're right and ofc I can do that, but I'm afraid of passing options that can totally break snippet compiler. I don't have knowledge of all possible settings but e.g. what will the compiler do with `-from-tasty` setting?

Reply:
> what will the compiler do with -from-tasty setting?

It will unpickle the .tasty files and compile them to .class files.